md2conf/converter.py

```diff
@@ -395,7 +395,7 @@ def _transform_link(self, anchor: ET._Element) -> Optional[ET._Element]:
 
         relative_path = os.path.relpath(absolute_path, self.base_path)
 
-        link_metadata = self.page_metadata.get(absolute_path)
+        link_metadata = self.page_metadata.get(absolute_path.resolve())
```

Comment:
We do invoke `pathlib.Path.resolve()` in `Application.synchronize()` but I agree that this is not sufficient given the use case you are describing. When concatenating relative paths starting with `..` to the document base path as we are transforming links in the Markdown document, we can end up with paths that are no longer canonical, and the code implicitly assumes that all paths are canonical, hence the error you have encountered.

Checking containment in the base path (few lines above) is a legitimate check, many applications do this for safety. However, when exporting a directory hierarchy, this base path should be the topmost directory we are exporting, rather than the document's location (as it is now), such that you can cross-reference documents in the directory hierarchy the way you describe.

I will look into possible solutions.